challenges.js

```diff
@@ -115,3 +115,45 @@ const challengesContainer = document.querySelector(".challenges-container");
 
 let view = new ChallengeListView();
 view.render(challengesContainer);
+
+//function keyword filter
+
+function keyFilter() {
+  // Declare variables
+  var input, filter, challenges, title, infoText, i;
+  input = document.getElementById('textFilter');
+  if (!input) {
+    console.error('Input element not found.!');
+    return;
+  }
+  filter = input.value.toUpperCase();
+  challenges = document.querySelectorAll('.challenges-container__challenge');
+
+  // loop through challenges
```

Comment:
This comment doesn't really provide any additional information to the reader. The next line of code is very clearly looping through something called `challenges`, so reading the code is actually easier than reading the comment.

The comment actually risks becoming misleading, if you ever make changes to the code and forget to change the comment, which is a common mistake.